cEP-0024.md

```diff
@@ -0,0 +1,107 @@
+
+## New Features
+
+### Auto-assign issue when it is mentioned in any of PR commits
+
+When a PR is opened in the repo and if any of its commit message
+mentions (Fixes, Closes) any unassigned issues and there is no CI failure, auto
+assign commit author to the mentioned unassigned issues. Comment on the
+PR (configurable) if someone else is working on the mentioned issue.
+
+### Auto-reject PR opened on master of forks
+
+Normally, contributors should make changes in different branch from master
+branch of the forked repo, and submit a PR. But, sometimes coala receives PRs
+from master branch of the forked repo. To solve this problem, coala needs
+mechanism by which GitMate automatically closes those PRs and comments on the
+PR informing the PR author to work on a different branch. Comment message will
+be configurable.
+
+### Auto-mark PR with necessary labels as per the reviews done
+
+When a reviewer reviews a PR, he/she either approves changes or request changes
+or comments something and leaves it there, but the PR would be still in the
+review queue. This will be a part of ack plugin. A setting to limit this plugin
+to certain teams will be added. GitMate will automatically mark labels as per
+the following conditions:
+
+- WIP label conditions:
+  1. Reviewer requests changes
+- Approve label conditions:
+  1. Reviewer approves changes
+  2. Anyone comments on the PR which has a CI failure
+
+### Fixes vs Closes automated checking
+
+Newcomers often confuse when they should use `Fixes` or `Closes` in a commit
+message. So, GitMate should automatically check every commit for issues which
+is being Fixed or Closed, if those issues have bug label (configurable),
+`Fixes` should be used, and vice versa.
+When the commit message contains 'Fixes', if the bug label is missing, the user
+will be informed to request (a maintainer) that the issue is updated to include
+the 'type/bug' label if they believe it is a bug they are fixing.
+
+When the commit message contains 'Closes', if the bug label is present, the
+user will be told that due to the presence of the bug label on the issue,
+they should use Fixes, or have the label removed.
+
+In both cases, remove 'pending review' label, mark 'wip' label. This will be
+part of ack plugin.
+
+### Assign Plugin
+
+Assign issue when a user comments on an issue requesting for an `assignment`
+to that issue. Will support settings for each team like limit team to work on
+issues with specific set of labels, limit number of issues of specified label
+a team member can work on, block assignment for certain labelled issues. corobo
+does all these things, but it is based just on Gitter messages and corobo is
+used in coala only, but GitMate is based on GitHub, used by many other users
+and is more configurable.
+
+```
+Settings:
+
+Blocked issues label: status/blocked
+
+Team - newcomer
+Label - newcomer  Limit - 1
+Label - low       Limit - 2
+
+Team - GCI
+Label - area/GCI  Limit - None
```

Comment:
area/GCI -> initiatives/gci

but this needs to be applied so that *no-one* in any *other* team can assign GCI issues.

Allowing the plugin to help with GCI is optional.
Breaking GCI is not acceptable.